### PR TITLE
created: card object

### PR DESCRIPTION
**Pull request by whatdoes3plus1equalsto**

### CODE REVIEW

**Review by VanJackal — CHANGES_REQUESTED** [review]
we will probably want a function in Card for creating the card from a card payload, probably `public static fromPayload(payload:CardPayload)`

**Comment by whatdoes3plus1equalsto** [comment]
> we will probably want a function in Card for creating the card from a card payload, probably `public static fromPayload(payload:CardPayload)`

changes made, but the code inheritance feels a bit weird, please have a look

File: src/objects/Card.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        return new CardCritter(
+            payload.cardid,
+            payload.name,
+            payload.playcost,
+            payload.rarity,
+            payload.image,
+            payload.description,
+            (payload.type_specific as { damage: number }).damage,
+            (payload.type_specific as { health: number }).health,
+            (payload.type_specific as { abilities: number[] }).abilities
```
Comment (VanJackal) [inline src/objects/Card.ts:81]: maybe cast this once to an intermediate object, or look into whether we could parameterize CardPayload so we'd have CardPayload<CardCritter> and CardPayload<CardItem> not sure if thats worth the effort though tbh

Comment (VanJackal) [reply src/objects/Card.ts]: should user the CardCritter type defined here: https://github.com/InternetEnemies/combatcritters-ts/blob/main/src/rest/payloads/cards.ts

Comment (whatdoes3plus1equalsto) [reply src/objects/Card.ts]: > should user the CardCritter type defined here: https://github.com/InternetEnemies/combatcritters-ts/blob/main/src/rest/payloads/cards.ts

updated

**Review by VanJackal — APPROVED** [review]

**Pull request merged**